weave/weave_client.py

```diff
@@ -817,3 +819,26 @@ def op_fn(*args, **kwargs):  # type: ignore
     op = Op(op_fn)
     op.name = name
     return op
+
+
+REDACT_KEYS = ("api_key",)
+REDACTED_VALUE = "REDACTED"
+
+
+def redact_sensitive_keys(obj: typing.Any) -> typing.Any:
+    if isinstance(obj, dict):
+        dict_res = {}
+        for k, v in obj.items():
+            if k in REDACT_KEYS:
+                dict_res[k] = REDACTED_VALUE
+            else:
+                dict_res[k] = redact_sensitive_keys(v)
+        return dict_res
+
+    elif isinstance(obj, list):
+        list_res = []
+        for v in obj:
+            list_res.append(redact_sensitive_keys(v))
+        return list_res
+
+    return obj
```

Comment:
do we also need a case for `isinstance(obj, tuple)`?

Reply:
added in https://github.com/wandb/weave/pull/1906